code/modules/mob/living/simple_animal/hostile/true_changeling.dm

```diff
@@ -0,0 +1,220 @@
+	var/datum/action/innate/changeling/devour/devour
+	var/datum/action/innate/changeling/spine_crawl/spine_crawl
+
+/mob/living/simple_animal/hostile/true_changeling/Initialize()
+	. = ..()
+	icon_state = "horror[rand(1, 5)]"
+	reform = new
+	reform.Grant(src)
+	devour = new
+	devour.Grant(src)
+	spine_crawl = new
+	spine_crawl.Grant(src)
+	for(var/mob/M in view(7, src))
+		flash_color(M, flash_color = list("#db0000", "#db0000", "#db0000", rgb(0,0,0)), flash_time = 5)
+	playsound(src, 'sound/creatures/rawrXD.ogg', 100, 1)
+
+/mob/living/simple_animal/hostile/true_changeling/Destroy()
+	QDEL_NULL(reform)
+	QDEL_NULL(devour)
+	QDEL_NULL(spine_crawl)
+	stored_changeling = null
+	return ..()
+
+/mob/living/simple_animal/hostile/true_changeling/Login()
+	. = ..()
+	to_chat(usr, playstyle_string)
+
+/mob/living/simple_animal/hostile/true_changeling/Life()
+	..()
+	adjustBruteLoss(-TRUE_CHANGELING_PASSIVE_HEAL) //True changelings slowly regenerate
+	time_spent_as_true++ //Used for re-forming
+	if(stored_changeling && time_spent_as_true >= TRUE_CHANGELING_FORCED_REFORM)
+		death() //After a while, the ling'll revert back without being able to control it
+
+/mob/living/simple_animal/hostile/true_changeling/Stat()
+	..()
+	if(statpanel("Status"))
+		if(stored_changeling)
+			var/time_left = TRUE_CHANGELING_FORCED_REFORM - time_spent_as_true
+			time_left = Clamp(time_left, 0, INFINITY)
+			stat(null, "Time Remaining: [time_left]")
+		stat(null, "Ignoring Gravity: [wallcrawl ? "YES" : "NO"]")
+
+/mob/living/simple_animal/hostile/true_changeling/death()
+	..(1)
+	new /obj/effect/gibspawner/human(get_turf(src))
+	if(stored_changeling && mind)
+		visible_message("<span class='warning'>[src] lets out a furious scream as it shrinks into its human form.</span>", \
+						"<span class='userdanger'>We lack the power to maintain this form! We helplessly turn back into a human...</span>")
+		stored_changeling.loc = get_turf(src)
+		mind.transfer_to(stored_changeling)
+		stored_changeling.Unconscious(300) //Make them helpless for some time
+		stored_changeling.status_flags &= ~GODMODE
+		qdel(src)
+	else
+		visible_message("<span class='warning'>[src] lets out a waning scream as it falls, twitching, to the floor.</span>", \
+						"<span class='userdanger'>We have fallen! We begin the revival process... (this will take around 45 seconds.)</span>")
+		spawn(450)
+			if(src)
+				visible_message("<span class='warning'>[src] reforms into a monster!</span>")
+				for(var/mob/M in view(7, src))
+					flash_color(M, flash_color = list("#db0000", "#db0000", "#db0000", rgb(0,0,0)), flash_time = 5)
+				new /obj/effect/gibspawner/human(get_turf(src))
+				revive() //Changelings can self-revive, and true changelings are no exception
+
+/mob/living/simple_animal/hostile/true_changeling/mob_negates_gravity()
+	return wallcrawl
+
+/mob/living/simple_animal/hostile/true_changeling/adjustFireLoss(amount)
+	playsound(src, 'sound/creatures/ling_scream.ogg', 100, 1)
```

Comment:
spam-prevention please.

Reply:
there's cooldowns on attacking and guns, it should be wailing a lot if people are

AND BEFORE ANYONE ASKS, I DID ADD IF(STAT)

Reply:
what about being on fire?

Reply:
Simplemobs can't be on fire but it didn't screech in space so

Reply:
> ling
> can't be on fire

KILL IT

serious note, +1 then

Reply:
more burn DAMAGE